server/controllers/__tests__/postQueryController.test.js

```diff
@@ -0,0 +1,45 @@
+const controller = require('../postQueryController')
+
+const parseIt = require('../../services/Parser')
+const postqueryFixtures = require('../__fixtures__/postQuery')
+const dateFixtures = require('../__fixtures__/dates')
+const mealtimeFixtures = require('../__fixtures__/mealtimes')
+const stopwordFixtures = require('../__fixtures__/stopwords')
+jest.mock('../../services/Parser')
+
+const res = {
+    json: jest.fn()
+}
+
+const log = jest.spyOn(global.console, "log").mockImplementation(() => {})
+
+beforeEach(() => {
+    jest.resetAllMocks()
+})
+
+describe('postquery controller test suite', ()=> {
+    const getDateMock = jest.fn()
+    const getMealtimeMock = jest.fn()
+    const getStopwordMock = jest.fn()
+    
+    test('should get the values provided in the query', () => {
+        const req = {query: {query: 'This is it i am not on lunch today!' }}
+        parseIt.mockImplementation(() => {
+            return {
+                getDate: getDateMock.mockReturnValue(dateFixtures),
```

Comment:
Try stepping through the code. 
This will cause every call to `getDate()` to return 
```json
[
    {
        "status": "success",
        "data": [
            {
                "value": "05/17/2021",
                "names": [
                    "today",
                    "now",
                    "currently"
                ]
            },
            {
                "value": "05/16/2021",
                "names": [
                    "yesterday"
                ]
            },
            {
                "value": "05/18/2021",
                "names": [
                    "tomorrow",
                    "currently"
                ]
            }
        ]
    }
]
```

then if you go into the controller you will see the `date` variable on line 25 has the following value
```
    {
      status: 'success',
      data: [
        { value: '05/17/2021', names: [Array] },
        { value: '05/16/2021', names: [Array] },
        { value: '05/18/2021', names: [Array] }
      ]
    }
```
But on line 26 you try to access the `type` and `value` properties when neither of those are defined at this level. 
I would take a look at the output of your code at this point and you fixtures as one of them isn't what you expect.